### PR TITLE
Final 

### PR DESCRIPTION
**Pull request by Uthmanbello**
## Highlights
- Setup React/Redux environment
- Fetched data from OpenWeather API
- Rendered the fetched data on the homepage
- Prepared navigation to the various pages 
- Styled the website to match the original design
- Fix errors using linters
- Publish the site using render.com
## Presentation
[Presentation link](https://www.loom.com/share/557c3d4066074f15b4021e9503fc7f3c)
## Deploy 
[Deploy link](https://weather-app-1unw.onrender.com/)

### CODE REVIEW

File: src/components/Navbar.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+          <ul>
+            {navlinks.map((navlink) => (
+              <li key={navlink.text}>
+                <NavLink to={navlink.path} style={({ isActive }) => ({ textDecoration: isActive ? 'underline' : '' })}>{navlink.text}</NavLink>
+              </li>
+            ))}
+          </ul>
+        </div>
+      </div>
+    </nav>
```
Comment (abdulhamiid) [inline src/components/Navbar.js]: Good job with your project. I really appreciate the effort you made to implement these features. 👍, Although, there are still some issues that you will need to fix, but you are almost there 🏃‍♂️

### Your Design
- [ ] Home page

image

- [ ] Details page

image


### Template

- [ ] Home page

image

- [ ] Details page

image

According to the project requirements, we should follow these [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), including:
   - Colors
   - Typography: font face, size, and weight.
   - Layout: `composition` and space between elements.

Please kindly let's add all the missing icons which are present in the design template as highlighted above 👆 

Also, let's consider using any of those icons that are used in the template for the developer info. 💪

File: src/setupTests.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+// jest-dom adds custom jest matchers for asserting on DOM nodes.
+// allows you to do things like:
+// expect(element).toHaveTextContent(/react/i)
+// learn more: https://github.com/testing-library/jest-dom
+import '@testing-library/jest-dom';
```
Comment (abdulhamiid) [inline src/setupTests.js:5]: I noticed that you have missed part of the requirements which is `Testing` 🤔. please kindly do that. 💪

According to the [project requirements ](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md)
image

Writing tests before and during writing the code is very useful. Tests in the application are reviewing the written code if it executes correctly and brings the required results. 

Learn more 👉 [Why testing a web application is important and how to test ReactJS application](https://www.linkedin.com/pulse/why-testing-web-application-important-how-test-reactjs-anna-danilec/)

File: src/components/Homes.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+            <FontAwesomeIcon
+              icon={faMapMarkerAlt}
+              style={{
+                color: 'rgb(53, 53, 165)', fontSize: '60px', marginBottom: '-80px', paddingTop: '15px',
+              }}
+            />
+            <span className="item-wrap">
+              Coordinates
+              <span className="item">
+                {Object.keys(data.coord).length}
```
Comment (abdulhamiid) [inline src/components/Homes.js]: From the project's requirement, It was mentioned that you should add another feature where you use a Filter stateless component to filter data retrieved from the API.

e.g You can implement that by adding a search bar where a user can type for example a category name he wants to get information from. I know you can do that 💪

image

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi @Uthmanbello 👋, 
Alt Text

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by anagudelogu — APPROVED** [review]
# STATUS: APPROVED! 🍾 🎆 🥇 

Hi @Uthmanbello, 
I'm Andres.

You did a great job working on this milestone, your project is complete! 🚀 
There is nothing else to say other than... it's time to merge it 🤖  
Congratulations! 🎉 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Great UI design. 💻 
✅ Descriptive PR title, summary and commit messages. 💬 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Uthmanbello** [comment]
@anagudelogu, 
Thanks a lot

**Pull request merged**